### ISSUE TITLE
Error in get_label

### ISSUE THREAD

**Issue by flyingpoops** [dmlc/xgboost#1323]
Hi, I am new to xgboost, and here is my problem:

When I assign large values to Dmatrix and try get_label, it returns a result with some margin of error. Is there any way to avoid this? Probably store label as a int?
Thanks in advance.

![qq 20160702173738](https://cloud.githubusercontent.com/assets/8256959/16542593/34ae79f8-407d-11e6-9d05-2e672a7533c4.jpg)

**Comment by bobye** [comment]
are you fitting categorical labels or numerical labels? If you work on categorical labels, how many categories you have? You could probably map those categorical index to smaller ones with the precision range of float32.

**Comment by flyingpoops** [comment]
Thanks for the reply, it is categorical labels, to be more precise, it ranges from 1.000016e+09 to 9.999932e+09. And I end up dividing it up and using LabelEncoder from sklearn.
